src/main/java/ru/practicum/shareit/item/repository/ItemRepositoryImpl.java

```diff
@@ -0,0 +1,83 @@
+package ru.practicum.shareit.item.repository;
+
+import org.springframework.stereotype.Repository;
+import ru.practicum.shareit.exceptions.MissingException;
+import ru.practicum.shareit.item.model.Item;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Repository
+public class ItemRepositoryImpl implements ItemRepository {
+
+    private int itemId = 0;
+    private final HashMap<Integer, Item> items = new HashMap<>();
```

Comment:
Лучше обозначать интерфейс, а уже инициализировать объект с определенной реализацией. Так программа становится более гибкой и не сломается, если нужно будет подменить реализацию на другую

Reply:
Map<Integer, Item> items = new HashMap<>()
```suggestion
    private final Map<Integer, Item> items = new HashMap<>();
```